pkg/scalers/aws_sigv4.go

```diff
@@ -0,0 +1,122 @@
+package scalers
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+// SigV4Config configures signing requests with SigV4.
+type SigV4Config struct {
+	Enabled bool   `yaml:"enabled,omitempty"`
+	Region  string `yaml:"region,omitempty"`
+}
+
+type awsAMPMetadata struct {
+	awsRegion        string
+	awsAuthorization awsAuthorizationMetadata
+}
```

Comment:
This struct is already available here: https://github.com/kedacore/keda/blob/main/pkg/scalers/aws_common.go#L28-L31

Reply:
Replaced with the one from aws_common.go as above